src/bytes.rs

```diff
@@ -1101,6 +1101,15 @@ unsafe fn release_shared(ptr: *mut Shared) {
     Box::from_raw(ptr);
 }
 
+fn ptr_map<F>(ptr: *mut u8, f: F) -> *mut u8
+where
+    F: FnOnce(usize) -> usize,
+{
+    let old_addr = ptr as usize;
+    let new_addr = f(old_addr);
+    ptr.wrapping_sub(old_addr).wrapping_add(new_addr)
+}
```

Comment:
take it or leave it: this is strange-looking enough that it seems like it _could_ be nice to have a comment explaining it?